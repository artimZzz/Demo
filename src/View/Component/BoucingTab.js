import React, { Component, PureComponent } from 'react';
import PropTypes from 'prop-types';
import BouncingView from './BoucingView';


export default class BouncingTab extends BouncingView {

    /* constructor */
    constructor(props) {
        super(props);

        this.inputRange = [0, 0.5, 1.0];
        this.outputRange = [1.0, 1.2, 1.0];
    }

    /* component */
    componentWillReceiveProps(nextProps) {
        if (nextProps.focused) {
            super.initAnimation();
        }
    }
}

BouncingTab.propTypes = {
    focused: PropTypes.bool
};
BouncingTab.defaultProps = {
    focused: false
};