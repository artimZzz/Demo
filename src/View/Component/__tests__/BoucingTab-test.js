// __tests__/BoucingTab-test.js
import React from 'react';
import BoucingTab from '../BoucingTab';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<BoucingTab />).toJSON();
    expect(tree).toMatchSnapshot();
});