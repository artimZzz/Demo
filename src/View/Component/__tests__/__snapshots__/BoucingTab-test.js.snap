// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "transform": Array [
        Object {
          "scaleX": 1,
        },
        Object {
          "scaleY": 1,
        },
      ],
    }
  }
/>
`;
